client/clients/router/request.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package router
+
+import (
+	"context"
+	"runtime/trace"
+	"sync"
+
+	"github.com/pingcap/errors"
+
+	"github.com/tikv/pd/client/opt"
+)
+
+// Request is a region info request.
+type Request struct {
+	requestCtx context.Context
+	clientCtx  context.Context
+
+	key         []byte
```

Comment:
Do we need to add some comments here, since the three fields represent three different requests? They cannot have values for a single request.